tests/sentry/workflow_engine/migration_helpers/test_migrate_rule.py

```diff
@@ -90,3 +97,84 @@ def test_create_issue_alert_no_actions(self):
             condition_result=True,
         ).exists()
         assert DataConditionGroupAction.objects.filter(condition_group=if_dcg).count() == 0
+
+    def test_update_issue_alert(self):
+        # TODO: update after action registry is merged
+        pass
+
+    def test_update_issue_alert_no_actions(self):
+        migrate_issue_alert(self.issue_alert, self.rpc_user)
+        conditions = [
+            {
+                "id": FirstSeenEventCondition.id,
+            },
+            {
+                "id": LatestReleaseFilter.id,
+            },
+        ]
+        payload = {
+            "name": "hello world",
+            "owner": Actor.from_id(user_id=self.user.id),
+            "environment": self.environment.id,
+            "action_match": "none",
+            "filter_match": "all",
+            "actions": [{"id": "sentry.rules.actions.notify_event.NotifyEventAction"}],
+            "conditions": conditions,
+            "frequency": 60,
+        }
+        update_migrated_issue_alert(self.issue_alert, payload, self.rpc_user)
+
+        issue_alert_workflow = AlertRuleWorkflow.objects.get(rule=self.issue_alert)
+        workflow = Workflow.objects.get(id=issue_alert_workflow.workflow.id)
+        assert workflow.name == payload["name"]
+        assert self.issue_alert.project
+        assert workflow.organization_id == self.issue_alert.project.organization.id
+        assert workflow.config == {"frequency": 60}
+
+        assert workflow.when_condition_group
+        assert workflow.when_condition_group.logic_type == DataConditionGroup.Type.NONE
+
+        assert len(RegionScheduledDeletion.objects.filter(model_name="DataCondition")) == 3
+
+        conditions = DataCondition.objects.filter(condition_group=workflow.when_condition_group)
```

Comment:
can we assert on the number of conditions and filters too? to check that they were overwritten properly?

Reply:
I omitted this since the conditions/filters that were scheduled for deletion would also be included in the count, but if we don't want to schedule the deletions anymore I can add this 👍🏼